RFCs/0009-Fn::For.md

```diff
@@ -0,0 +1,903 @@
+# New Intrinsic Function for looping: `Fn::Map`
+
+* **Original Author(s):**: @mingujo
+* **Tracking Issue**: [Tracking Issue: Fn::Map](https://github.com/aws-cloudformation/cfn-language-discussion/issues/9)
+
+# Summary
+We are proposing to introduce a new intrinsic function called `Fn::Map`. It will allow users to iterate over a given list to replicate a particular template fragment thereby avoiding repetitively writing a similar declaration for each object (e.g. Resource, Output, or Resource Properties).
+
+`Fn::Map` traverses a single given List of Strings from first to last, and over each iteration, it references each item under a given template snippet to be replicated.
+
+We additionally introduce `Fn::Merge` which can be used to merge the result of Fn::Map into its parent object.
+# Motivation
+
+In a CloudFormation template, a single Resource configures one infrastructure object. Therefore, a template can become verbose when you manually declare multiple similar resources. Some examples would include a pool of EC2 instances with the same configurations but different instance types, or S3 Bucket Notification configurations with different SNS Topics.
+
+# Declaration
+## Fn::Map
+### Json
+```json
+{
+  "Fn::Map": {
+    "index": "Index",
```

Comment:
The parameter names are lowercase, but basically everything in CloudFormation is UpperCase, from API parameters to template sections to intrinsic function names. I'd suggest UpperCase for consistency.

Reply:
I agree. Thanks for raising this.